### ISSUE TITLE
508-defect-3 [COGNITION] Direct Deposit Component - Header should be updated when users switch into edit mode

### ISSUE THREAD

**Issue by 1Copenut** [department-of-veterans-affairs/va.gov-team#18436]
# [508-defect-3](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-3)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

## Point of Contact

**VFS Point of Contact:** _Trevor_

## User Story or Problem Statement

As a screen reader user, I want to hear that I have switched into the edit inline "state" of the direct deposit component.

## Details

Screen readers do not have any aural cues that users have entered or exited the edit in place mode. We have a state Boolean `editing` that could be used to set the component header conditionally. Screenshot attached below.

## Acceptance Criteria

- [ ] Component header says "Edit direct deposit information" when editing inline
- [ ] Component header says "Direct deposit information" when read-only
- [ ] Zero axe-core violations
- [ ] Focus is managed properly as noted in #18435 

## WCAG or Vendor Guidance (optional)

* [Info and Relationships: Understanding SC 1.3.1](https://www.w3.org/TR/UNDERSTANDING-WCAG20/content-structure-separation-programmatic.html)

## Screenshots or Trace Logs

![Screen Shot 2021-01-12 at 4 02 24 PM](https://user-images.githubusercontent.com/934879/104379863-b6757500-54ef-11eb-87a9-5803357c102f.png)
